src/backend/distributed/utils/colocation_utils.c

```diff
@@ -1194,8 +1197,8 @@ ColocatedTableId(int32 colocationId)
 	HeapTuple heapTuple = systable_getnext(scanDescriptor);
 	while (HeapTupleIsValid(heapTuple))
 	{
-		bool isNullArray[Natts_pg_dist_partition];
-		Datum datumArray[Natts_pg_dist_partition];
+		Datum* datumArray = (Datum *) palloc(tupleDescriptor->natts * sizeof(Datum));
```

Comment:
I'd allocate these outside & right before of the loop to avoid excessive amount of palloc calls.